### DISCUSSION TITLE
Bitcoin Core / Research the history of Core GUI changes through Github issues

### DISCUSSION THREAD

**Discussion by johnsBeharry** [BitcoinDesign/Meta#19] [General]
### Abstract
In order to start a redesign of Bitcoin Core GUI, we should try to understand design constraints, and technical decisions that have lead to its current state.

The way we do this would be to start gathering links to issues that may be relevant to any redesign efforts so designers coming in do not make assumptions and waste time on things that are not practical, or have already explored.

### Motivation
We would need to come up with a strategy to get designers involved in the ongoing UI/UX decisions made to the GUI, and find out if it's practical to make more substantial improvements to the UI, and understanding how to go about doing so.

### Issues
- https://github.com/bitcoin/bitcoin/pull/16883
- https://github.com/bitcoin-core/gui/issues/24
- https://github.com/bitcoin/bitcoin/pull/14810#issuecomment-635804151

---

### REF

- [Bitcoin Core GUI / Design Jam Notes](https://hackmd.io/XdI-wmBMRES96HL8x_KDCQ)
- [Discussion on Slack](https://bitcoindesign.slack.com/archives/C0174N5KUF9/p1596267094004700)

**Comment by michaelfolkson** [comment]
I think this is a good place for collecting together all links on past discussions, PRs, Issues on attempts to implement automated GUI testing to work out how we can move the testing conversation forward.

**Comment by johnsBeharry** [comment]
### New GUI features

Every new feature gets implemented as a RPC call, then after about 2 releases (once stability is confirmed) it is implemented in the GUI.

- Major releases are made twice a year(estimate)
- GUI doesn't have a large testing framework, so continuous manual testing is done across platforms by developer and reviewers.
- RPC has a automated testing coverage.
- Lack of a certain GUI feature which is implemented in the RPC may just be down to the issue not getting design and developer attention

### Hardware Wallet Support
- https://github.com/bitcoin/bitcoin/issues/14145
- https://github.com/bitcoin-core/HWI#readme

- [ ] Lookup references for hardware wallet support

---

_Notes from Bitcoin Core GUI call BitcoinDesign/Guide#2 https://github.com/BitcoinDesign/Meta/issues/13_

**Comment by Bosch-0** [comment]
## Writing Copy
[Optech Style Guide](https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md) - These are by no means industry wide standards but they seem to be pretty consistent with what most people use. A standard style guide for bitcoin doesn't (or maybe should never?) currently exist. 

## Font discussions

- https://github.com/bitcoin-core/gui/pull/79
- https://github.com/bitcoin/bitcoin/pull/16883
- https://github.com/bitcoin/bitcoin/pull/17311#issuecomment-550453875 - more specifically discusses issues with OFL licensing. 
- https://github.com/bitcoin/bitcoin/pull/16432#issuecomment-514486077 - issue with there being no monospace font inherited with Qt Widgets on MacOS
- https://github.com/bitcoin/bitcoin/pull/18313

## Hardware wallet

https://github.com/bitcoin-core/gui/pull/4 Needs some designer reviewing this PR. 

## Pruned nodes 

Enable wallet import on pruned nodes - https://github.com/bitcoin/bitcoin/pull/16037 currently importing wallets that have txs in the blocks you prune will not be shown. This is a UX issue that needs to be overcome. This bitcointalk post goes in more depth on this topic - https://bitcointalk.org/index.php?topic=5213552.0 

## Why Bitcoin Core doesn't support BIP 39 

https://github.com/bitcoin/bitcoin/pull/6265
https://bitcoin.stackexchange.com/questions/88237/is-there-a-reason-to-why-bitcoin-core-does-not-implement-bip39
importing may be coming in the future though - https://github.com/bitcoin/bitcoin/issues/16393 https://github.com/bitcoin/bitcoin/issues/19151

**Comment by hebasto** [comment]
Please note that https://github.com/bitcoin-core/gui/pull/79 is actually a (visual) bug fix. Such PRs (bug fixes) usually get more reviewers' attention as they are justified by their nature.

Any other improvement or feature PR require strong motivation, and I think having the (draft) Designer Guide could be a such motivation.

**Comment by jonatack** [comment]
> Every new feature gets implemented as a RPC call, then after about 2 releases (once stability is confirmed) it is implemented in the GUI.

This seems generally true. Yet open source is an ad-hoc, scratch-your-own-itch process and in practice I have the impression there are quite a few exceptions that can depend on the initiative of each contributor. Two examples I've been involved with, for an idea:

- the GUI has displayed the number of in and out peer connections for years; the same via the RPC (`getnetworkinfo`) and a CLI option (`-getinfo`) was only just now merged into Bitcoin Core master for the next release in November

- the experimental Asmap feature info was first exposed via both the RPC (`getpeerinfo`) and the GUI (peers detail window) in the same release

That said, the general concepts you've expressed are very good.

**Comment by danielnordh** [comment]
Is this issue still relevant?
@Bosch-0 has been doing work elsewhere on Core UI.

**Reply by johnsBeharry** [reply]
Good case to utilise the discussion feature.
